src/resource.js

```diff
@@ -397,6 +422,27 @@ angular.module('hypermedia')
         if (profileUris) this.$profile = profileUris;
 
         return this;
+      }},
+
+      /**
+       * Merges the resource with new data following algorithm defined
+       * in JSON Merge Patch specification (Rfc 7386, https://tools.ietf.org/html/rfc7386).
+       *
+       * @function
+       * @param {object} data
+       * @param {object} [links]
+       * @returns the resource
+       */
+      $merge: {value: function (data){
+        var self = this;
+        Object.keys(data).forEach(function(key){
+          if (data[key] === null) {
+            delete self[key];
+          } else {
+            self[key] = data[key];
+          }
+          return self;
+        });
```

Comment:
This algorithm does not do a deep merge of objects [as described in the RFC](https://tools.ietf.org/html/rfc7386#section-2). It probably should.